tests/test_gradient_boosting.py

```diff
@@ -171,3 +172,55 @@ def test_early_stopping_classification(data, scoring, validation_split, tol):
         assert n_iter_no_change <= gb.n_iter_ < max_iter
     else:
         assert gb.n_iter_ == max_iter
+
+
+# TODO: Remove if / when numba issue 3569 is fixed.
+def custom_check_estimator(Estimator):
+    # Same as sklearn.check_estimator, skipping tests that can't succeed.
+
+    from sklearn.utils.estimator_checks import _yield_all_checks
+    from sklearn.utils.testing import SkipTest
+    from sklearn.exceptions import SkipTestWarning
+    from sklearn.utils import estimator_checks
+
+    estimator = Estimator
+    name = type(estimator).__name__
+
+    for check in _yield_all_checks(name, estimator):
+        if (check is estimator_checks.check_fit2d_1feature or
+                check is estimator_checks.check_fit2d_1sample):
+            # X is both Fortran and C aligned and numba can't compile.
+            # Opened numba issue 3569 (not sure if this is by design)
+            continue
```

Comment:
In the meantime, we could make the `GradientBoostingMachine` classes raise a `ValueError` with an informative error message.